packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -957,6 +964,7 @@ export class DatabaseInstance extends DatabaseInstanceSource implements IDatabas
       masterUsername: credentials.username,
       masterUserPassword: credentials.password?.toString(),
       storageEncrypted: props.storageEncryptionKey ? true : props.storageEncrypted,
+      ...(props.publiclyAccessible != undefined && { publiclyAccessible: props.publiclyAccessible }),
```

Comment:
would rather do this in `this.newCfnProps`:

https://github.com/aws/aws-cdk/blob/23d10dfa23d597d9a42160f6f579ca3d3b49ac14/packages/%40aws-cdk/aws-rds/lib/instance.ts#L707

where we can just do
`publiclyAccessible: props.publiclyAccessible ?? this.vpcPlacement && this.vpcPlacement.subnetType === ec2.SubnetType.PUBLIC`